iaso/api/org_unit_change_requests/views.py

```diff
@@ -20,8 +27,16 @@
 class OrgUnitChangeRequestViewSet(
     CreateModelMixin, ListModelMixin, RetrieveModelMixin, UpdateModelMixin, viewsets.GenericViewSet
 ):
-    permission_classes = [HasOrgUnitsChangeRequestPermission]
+    filter_backends = [filters.OrderingFilter, django_filters.rest_framework.DjangoFilterBackend]
     filterset_class = OrgUnitChangeRequestListFilter
+    pagination_class = OrgUnitChangeRequestPagination
+
+    def get_permissions(self):
+        if self.action == "partial_update":
+            permission_classes = [HasOrgUnitsChangeRequestReviewPermission]
+        else:
+            permission_classes = [HasOrgUnitsChangeRequestPermission]
+        return [permission() for permission in permission_classes]
```

Comment:
I may have misunderstood here but I believe this listing is only relevant for reviewer so I would change it to:
```suggestion
    permission_classes = [HasOrgUnitsChangeRequestReviewPermission]
```

Reply:
No we can't do that because we are in the context a viewset and the other views (`list`, `update`, etc.) only require `HasOrgUnitsChangeRequestPermission`.

Reply:
You are right, I forgot that the `POST` and `PUT` were used by regular users. I would change it like so:
```suggestion
    def get_permissions(self):
        if self.action == "insert":
            permission_classes = [HasOrgUnitsChangeRequestPermission]
        else:
            permission_classes = [HasOrgUnitsChangeRequestReviewPermission]
        return [permission() for permission in permission_classes]
```
There is no need for "regular" users to list all the change requests. They need only to create new ones.

Reply:
This wouldn't work because `insert` is not a [valid action on a viewset](https://www.django-rest-framework.org/api-guide/viewsets/#viewset-actions).

My code is actually correct and conform to [what we said in IA-2536](https://bluesquare.atlassian.net/browse/IA-2536).

Trust me on this one ;)

Reply:
I wrote pseudocode, in the sense that I didn't check the possible actions, I thought the intention was clear as is.
If you want, you can replace `insert` by `create`.

Anyway, I'm not arguing whether your code is correct or wrong. I'm just asking if it makes sense to expose `list`, `retrieve`,  `destroy`, and `update` to regular users since we only need them to create new change requests and not act upon them.

Reply:
> I'm just asking if it makes sense to expose list, retrieve,  destroy, and update to regular users since we only need them to create new change requests and not act upon them.

That's what we agreed on in [IA-2536](https://bluesquare.atlassian.net/browse/IA-2536). Shall we change that?

Reply:
That's what I'm asking you :slightly_smiling_face:

Reply:
Let's stick to what we agreed on.

Reply:
As discussed, I believe it's "dangerous" to let `update` and `delete` open. I also don't think it makes much sense to leave `list` and `retrieve` accessible, but they are not a threat.
Anyway, I have raised the concern, it's not my decision, the code is fine by me.